### PR TITLE
integration/apiserver/:Remove direct prometheus dependencies from k/k

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
 
 
/kind cleanup 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/89267

removing direct prometheus dependency in test/integration/apiserver/

/cc @serathius
/cc @logicalhan

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by logicalhan** [comment]
It looks like golint is failing.

File: staging/src/k8s.io/component-base/metrics/testutil/testutil.go [hunk 1]
```diff
@@ -68,3 +69,21 @@ func NewFakeKubeRegistry(ver string) metrics.KubeRegistry {
...
+	}
+	for _, metric := range samples {
+		switch name := string(metric.Metric[model.MetricNameLabel]); name {
+		case metricName:
+			concurrency[string(metric.Metric[model.LabelName(labelPriorityLevel)])] = int(metric.Value)
+		}
+	}
+
+	return concurrency, nil
+}
```
Comment (logicalhan) [inline staging/src/k8s.io/component-base/metrics/testutil/testutil.go:89]: This is a very specific function for (what it sounds like) a very specific metric.. why are you extracting this to testutils?

Comment (tanjunchen) [reply staging/src/k8s.io/component-base/metrics/testutil/testutil.go]: 1. remove dependency
2. I don't find that similar function in staging/src/k8s.io/component-base/metrics/testutil/
the similar function is 
```
// GetMetricValuesForLabel returns value of metric for a given dimension
func GetMetricValuesForLabel(ms Metrics, metricName, label string) map[string]int64 {
	samples, found := ms[metricName]
	result := make(map[string]int64, len(samples))
	if !found {
		return result
	}
	for _, sample := range samples {
		count := int64(sample.Value)
		dimensionName := string(sample.Metric[model.LabelName(label)])
		result[dimensionName] = count
	}
	return result
}
```

3. I guess that we can have this function in testutil for (input ,metricName, label ) to extract value

```
// ExtractMetricSamples parses the prometheus metric samples from the input string.
func ExtractMetricSamples(metricsBlob string) ([]*model.Sample, error) {
	dec := expfmt.NewDecoder(strings.NewReader(metricsBlob), expfmt.FmtText)
	decoder := expfmt.SampleDecoder{
		Dec:  dec,
		Opts: &expfmt.DecodeOptions{},
	}

	var samples []*model.Sample
	for {
		var v model.Vector
		if err := decoder.Decode(&v); err != nil {
			if err == io.EOF {
				// Expected loop termination condition.
				return samples, nil
			}
			return nil, err
		}
		samples = append(samples, v...)
	}
}
```
Or do you have other good ideas？

Comment (logicalhan) [reply staging/src/k8s.io/component-base/metrics/testutil/testutil.go]: Yes, I think I prefer just extracting out `ExtractMetricSamples` into the metrics directory. I believe you will have to create a wrapper around the `prometheus.Sample` struct if we don't have one for it already. (@RainbowMango probably knows better than I at this point)

Comment (tanjunchen) [reply staging/src/k8s.io/component-base/metrics/testutil/testutil.go]: thanks . i will think more about it.

Comment (tanjunchen) [reply staging/src/k8s.io/component-base/metrics/testutil/testutil.go]: /cc @logicalhan 
how about GetMetricValueFromSample , it is more common.

**Pull request closed without merging**